src/fragment-components/a-radio-group.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import {
+	Dropdown,
+	TextInput,
+	RadioButtonGroup
+} from 'carbon-components-react';
+import { AComponent, ComponentInfo } from './a-component';
+import { ComponentCssClassSelector } from '../components/css-class-selector';
+import image from './../assets/component-icons/radio-group.svg';
+import {
+	angularClassNamesFromComponentObj,
+	nameStringToVariableString,
+	reactClassNamesFromComponentObj
+} from '../utils/fragment-tools';
+
+export const ARadioGroupStyleUI = ({ selectedComponent, setComponent }: any) => {
+	const orientationItems = [
+		{ id: 'horizontal', text: 'Horizontal' },
+		{ id: 'vertical', text: 'Vertical' }
+	];
+
+	const labelPositions = [
+		{ id: 'left', text: 'Left' },
+		{ id: 'right', text: 'Right' }
+	];
+
+	return <>
+		<TextInput
+			value={selectedComponent.legend}
+			labelText='Legend name'
+			placeholder='Legend name'
+			onChange={(event: any) => {
+				setComponent({
+					...selectedComponent,
+					legend: event.currentTarget.value
+				});
+			}}
+		/>
+		<Dropdown
+			label='Orientation'
+			titleText='Orientation'
+			items={orientationItems}
+			initialSelectedItem={orientationItems.find(item => item.id === selectedComponent.orientation)}
+			itemToString={(item: any) => (item ? item.text : '')}
+			onChange={(event: any) => setComponent({
+				...selectedComponent,
+				orientation: event.selectedItem.id
+		})}/>
+		<Dropdown
+			label='Label position'
+			titleText='Label position'
+			items={labelPositions}
+			initialSelectedItem={labelPositions.find(item => item.id === selectedComponent.labelPosition)}
+			itemToString={(item: any) => (item ? item.text : '')}
+			onChange={(event: any) => setComponent({
+				...selectedComponent,
+				labelPosition: event.selectedItem.id
+		})}/>
+		<ComponentCssClassSelector componentObj={selectedComponent} setComponent={setComponent} />
+	</>;
+};
+
+export const ARadioGroupCodeUI = ({ selectedComponent, setComponent }: any) => {
+	const allItems = selectedComponent.items.map((item: any) => ({
+		text: item.labelText,
+		id: item.id,
+		defaultChecked: item.defaultChecked
+	}));
+	allItems.push({ text: 'None', id: 'none', defaultChecked: '' });
+	selectedComponent.defaultSelected = selectedComponent.items.find(((item: any) => item.defaultChecked))?.id;
```

Comment:
I just noticed...

This is not the way to update values in React and leads to all sorts of bugs. Any updates should happen using `setComponent` instead.

Why is this even happening in the code UI? 😕

Reply:
because the default selection is updated through the code UI panel

Reply:
updated using setComponent